platform.txt

```diff
@@ -158,9 +158,9 @@ tools.esptool.upload.params.quiet=
 # First, potentially perform an erase or nothing
 # Next, do the binary upload
 # Combined in one rule because Arduino doesn't support upload.1.pattern/upload.3.pattern
-tools.esptool.upload.pattern="{cmd}" "{runtime.platform.path}/tools/upload.py" --chip esp8266 --port "{serial.port}" --baud "{upload.speed}" "{upload.verbose}" {upload.erase_cmd} {upload.resetmethod} write_flash 0x0 "{build.path}/{build.project_name}.bin"
+tools.esptool.upload.pattern="{cmd}" "{runtime.platform.path}/tools/upload.py" -E --chip esp8266 --port "{serial.port}" --baud "{upload.speed}" "{upload.verbose}" {upload.erase_cmd} {upload.resetmethod} write_flash 0x0 "{build.path}/{build.project_name}.bin"
 
-tools.esptool.upload.network_pattern="{network_cmd}" "{runtime.platform.path}/tools/espota.py" -i "{serial.port}" -p "{network.port}" "--auth={network.password}" -f "{build.path}/{build.project_name}.bin"
+tools.esptool.upload.network_pattern="{network_cmd}" "{runtime.platform.path}/tools/espota.py" -E -i "{serial.port}" -p "{network.port}" "--auth={network.password}" -f "{build.path}/{build.project_name}.bin"
```

Comment:
`{cmd} espota.py -E` -> `{cmd} -E -- espota.py`?
assuming cmd expands to python.exe, arg is intended for python
(and probably also applies above, where python cmd is spelled out)

Reply:
Yes, I will update soon. That’s what I get for submitting PRs at 2am…

Reply:
Done with latest push.  I was worried about `-I` and the scripts directory, but we don't have any cross-script dependencies so it does work fine.

For PlatformIO, it seems the builder script doesn't use the packages Python and relies on `$PYTHONEXE` which I assume is part of/required for Platform.IO.

Reply:
True, lgtm. (although, I'd still add `--` for future's sake)

PIO itself needs system / bundled python to run, so it is slightly different there. e.g. with esptool, it just downloads a standalone script, since it already includes pyserial as it's core dependency